api/envoy/config/tap/v3/common.proto

```diff
@@ -101,6 +107,30 @@ message HttpHeadersMatch {
   repeated route.v3.HeaderMatcher headers = 1;
 }
 
+// HTTP generic body match configuration.
+// List of text strings and hex strings to be located in HTTP body.
+// All specified strings must be found in the HTTP body for positive match.
+// The search may be limited to specified number of bytes from the body start.
+message HttpGenericBodyMatch {
```

Comment:
I think it's worth some comments about complexity here and in the usage docs, maybe a "use with care" warning block?  It's obvious if you think about it but if you have a large post body this going to be pretty expensive, and gets more expensive fast if you've got longer (partially matching) strings or many strings.

Reply:
Agreed. I will add comments about complexity. But unfortunately entire body must be scanned at least once per specified pattern.

Reply:
yeah, I think many people would want to prefix match, and won't realize they should set the length to length(prefix) to avoid scanning the full body of large posts over and over.  maybe a call-out for that too?

Reply:
Added `attention `section to docs describing that search is potentially cpu intensive and if expected pattern's location is known, search should be limited to specified number of bytes.